apps/omg_watcher/lib/exit_processor.ex

```diff
@@ -294,6 +305,27 @@ defmodule OMG.Watcher.ExitProcessor do
     {:reply, canonicity_result, state}
   end
 
+  def handle_call({:create_challenge, Utxo.position(blknum, txindex, oindex) = exiting_utxo_pos}, _from, state) do
+    with spending_blknum_response <- exiting_utxo_pos |> Utxo.Position.to_db_key() |> OMG.DB.spent_blknum(),
+         %{txhash: txhash} <- OMG.Watcher.DB.Transaction.get_by_position(blknum, txindex),
+         {:ok, exit_id} <- OMG.Eth.RootChain.get_standard_exit_id(txhash, oindex),
+         {:ok, raw_spending_proof, exit_info} <-
+           Core.get_challange_data(spending_blknum_response, exiting_utxo_pos, state) do
+      spending_proof =
+        case raw_spending_proof do
+          blknum when is_number(blknum) ->
+            {:ok, hashes} = OMG.DB.block_hashes([blknum])
+            {:ok, [spending_block]} = OMG.DB.blocks(hashes)
+            spending_block
+
+          signed_tx ->
+            signed_tx
+        end
+
+      {:reply, {:ok, Core.create_challenge(exit_info, spending_proof, exiting_utxo_pos, exit_id)}, state}
```

Comment:
why assume that Core.create_challenge never return error?

Reply:
Here we're just passing information further about errors